src/rrdp-iostat/rrdp_iostat.ml

```diff
@@ -335,14 +319,16 @@ module Blktap3_stats_wrapper = struct
     in
 
     Mutex.execute m (fun () ->
-        Hashtbl.fold (fun (domid, devid) pid_opt acc ->
-            match pid_opt with
-            | Some pid -> ((domid, devid), pid)::acc
+        Hashtbl.fold (fun (domid, devid) stats_opt acc ->
+            match stats_opt with
+            | Some stats -> ((domid, devid), Blktap3_stats.copy stats)::acc
             | None ->
               match pid_from_xs (domid, devid) with
               | Some pid ->
-                Hashtbl.replace cache (domid, devid) (Some pid);
-                ((domid, devid), pid)::acc
+                let stat_file = Printf.sprintf "%s/td3-%d/vbd-%d-%d" shm_devices_dir pid domid devid in
+                let stats = Blktap3_stats.of_file stat_file in
```

Comment:
This file _may_ not be present yet, even if the `vbd3-x-y` file is there, and we have an entry in the cache. Can this therefore blow up here?